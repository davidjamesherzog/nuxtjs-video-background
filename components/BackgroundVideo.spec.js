import { mount } from '@vue/test-utils';
// import BackgroundVideo from '@/components/BackgroundVideo.vue';
import {
  // eslint-disable-next-line import/no-named-default
  default as BackgroundVideo,
  __RewireAPI__ // eslint-disable-line import/named
} from '@/components/BackgroundVideo.vue';

describe('BackgroundVideo', () => {
  describe('data', () => {
    test('returns defaults', () => {
      expect(BackgroundVideo.data()).toEqual({
        video: null
      });
    });
  });

  describe('mounted', () => {
    test('should set create video.js instance and start video', () => {
      const mockContext = {
        src: 'video.mp4'
      };
      const mockVideo = {
        src: jest.fn()
      };
      const spy = jest.fn(() => mockVideo);
      __RewireAPI__.__Rewire__('videojs', spy);
      BackgroundVideo.mounted.bind(mockContext)();

      expect(spy).toHaveBeenCalledWith('video-player', {
        autoplay: true,
        loop: true,
        muted: true,
        controls: false,
        techOrder: ['html5']
      });
      expect(mockContext.video).toBeDefined();
      expect(mockContext.video.src).toHaveBeenCalledWith([
        {
          type: 'video/mp4',
          src: mockContext.src
        }
      ]);
    });
  });

  describe('templates', () => {
    test('is a Vue instance', () => {
      const wrapper = mount(BackgroundVideo, {
        propsData: {
          src: 'some.mp4'
        }
      });
      expect(wrapper.vm).toBeTruthy();
      expect(wrapper.element).toMatchSnapshot();
    });

    test('should have video element', () => {
      const wrapper = mount(BackgroundVideo, {
        propsData: {
          src: 'some.mp4'
        }
      });
      const video = wrapper.find('[data-background-video]');

      expect(video.exists()).toBeTruthy();
      expect(video.classes()).toContain('video-js');
      expect(video.classes()).toContain('vjs-theme-sea');
      expect(wrapper.element).toMatchSnapshot();
    });
  });
});
