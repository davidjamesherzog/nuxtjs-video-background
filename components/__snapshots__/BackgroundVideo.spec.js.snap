// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackgroundVideo templates is a Vue instance 1`] = `
<video
  class="video-js vjs-theme-sea"
  data-background-video=""
  id="video-player"
/>
`;

exports[`BackgroundVideo templates should have video element 1`] = `
<video
  class="video-js vjs-theme-sea"
  data-background-video=""
  id="video-player"
/>
`;
